macro/makeGenieEvents.py

```diff
@@ -134,9 +107,35 @@ def addHists():
   genie_interface.add_hists(neutrinos, "genie-"+sDict[p]+".root", p)
   os.chdir('../')
 
-if (MakeSpline):
+if ("splinedir" not in args):
```

Comment:
I don't understand this: `splinedir` should always be in args, as it has a default value. Or is it undefined, if the subparser is not chosen?
Is there no better way to check whether we are in the spline or sim subparser?

Reply:
If the subparser is not chosen, all its args are not defined. I confirmed it by inspecting the args namespace when the two different options are called.

An alternative is to save the subparser in args itself, as discussed here:
https://stackoverflow.com/questions/8250010/argparse-identify-which-subparser-was-used/9286586#9286586

However, this would lead back to the previous case, with an if(args.comment=="spline") statement (equivalent to if(MakeSpline)).

Reply:
Thank you for the explanation!  I agree that your check is probably the best way to do this :)